contracts/tests/test_unruggable_memecoin.cairo

```diff
@@ -45,3 +45,324 @@ fn test_mint() {
     let balance = safe_dispatcher.balance_of(owner);
     assert(balance == initial_supply, 'Invalid balance');
 }
+
+
+#[test]
+fn test_transfer() {
+    let owner = contract_address_const::<42>();
+    let recipient = contract_address_const::<43>();
+    let initial_supply = 1000.into();
+    let contract_address = deploy_contract(
+        owner, owner, 'UnruggableMemecoin', 'MT', initial_supply
+    );
+
+
+    // set owner as caller
+    start_prank(CheatTarget::One(contract_address), owner);
+
+    // make transfer 
+    let safe_dispatcher = IUnruggableMemecoinDispatcher { contract_address };
+    safe_dispatcher.transfer(recipient, initial_supply - 1);
+
+    // check owner's balance. Should be equal to initial supply.
+    let owner_balance = safe_dispatcher.balance_of(owner);
+    assert(owner_balance == 1, 'invalid owner balance');
+
+    // check initial balance. Should be equal to initial supply.
+    let recipient_balance = safe_dispatcher.balance_of(recipient);
+    assert(recipient_balance == initial_supply - 1 , 'invalid recipient balance');
+}
+
+
+#[test]
+fn test_transfer_2() {
+    /// Ensure that transfers can be made out to 
+    /// up to `MAX_HOLDERS_BEFORE_LAUNCH` addresses
+
+    let owner = contract_address_const::<42>();
+    let initial_supply = 1000.into();
+    let contract_address = deploy_contract(
+        owner, owner, 'UnruggableMemecoin', 'MT', initial_supply
+    );
+
+
+    // set owner as caller
+    start_prank(CheatTarget::One(contract_address), owner);
+
+    let safe_dispatcher = IUnruggableMemecoinDispatcher { contract_address };
+
+
+    // index starts from 1 because the owner
+    // is considered to be the first hodler
+    let mut index = 1; 
+    loop {
+        if index == MAX_HOLDERS_BEFORE_LAUNCH {
```

Comment:
it should be 
```suggestion
        if index > MAX_HOLDERS_BEFORE_LAUNCH {
```

It works because the `mint` functoin doesn't increase holders_count, which it should - this will be solved once the "unruggable" transfer mechanism is implemented in the `_transfer`

Reply:
ah i see. made it that way initially then removed it but forgot to update test
- mint and transfer now increase holder count